### PR TITLE
Update the distributed throttling page to use binary publisher

### PR DESCRIPTION
**Pull request by VirajSalaka**
### Purpose
Add "how to" documentation for Distributed Throttling.

### Issues

Fixes #

---
#### Maintainers: Check before merge
- [x] Manual verification

### CODE REVIEW

File: en/docs/how-tos/rate-limiting/distributed-throttling.md [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
-
-The API Microgateway upon recieving a request, checks against the local counter and if throttling limit  has not exceeded it publishes the events via a stream to a central traffic management solution. This is done over HTTP. The  central traffic management solution then  executes throttle policies against the events streams. When a particular request is throttled, the  central traffic management solution sends the details of the throttled out event to a JMS topic. Each API Microgateway node is subscribed to this JMS topic, and updates the local counter when the JMS topic is updated.  Hence the API Microgateway nodes gets notified of the throttle decisions through JMS messages.
-
-![distributed throttling]({{base_path}}/assets/img/how-tos/distributed-throttling.png)
+WSO2 API Microgateway has an in-memory mechanism by default, to handle throttling
+([node-level throttling]({{base_path}}/how-tos/rate-limiting/adding-throttling-policies)).
```
Comment (praminda) [inline en/docs/how-tos/rate-limiting/distributed-throttling.md:4]: I've noticed if we define urls in format `{{base_path}}/<path>` it always open the link in a new tab assuming its a external link. however if we omit `{{base_path}}` and keep only the absolute path to the linked page (from site root `/`) it opens in the same tab.
```suggestion
([node-level throttling](/how-tos/rate-limiting/adding-throttling-policies)).  
```

File: en/docs/how-tos/rate-limiting/distributed-throttling.md [hunk 2]
```diff
@@ -37,25 +55,131 @@ The API Microgateway upon recieving a request, checks against the local counter
...
+            </tr>
+            <tr class="odd">
+                <td><code>jmsConnectionPassword</code></td>
+                <td><code>""</code></td>
+                <td>The password used to establish  the message broker connection</td>
+            </tr>
+        </tbody>
+    </table>
+    
+    The message broker connection URL. For e.g. a [WSO2 API instance can be used as the Traffic Manager](https://docs.wso2.com/display/AM260/Product+Profiles).
```
Comment (praminda) [inline en/docs/how-tos/rate-limiting/distributed-throttling.md:97]: Shall we point to 320 apim docs?

File: en/docs/how-tos/rate-limiting/distributed-throttling.md [hunk 3]
```diff
@@ -37,25 +55,131 @@ The API Microgateway upon recieving a request, checks against the local counter
...
+            </tr>
+            <tr class="odd">
+                <td><code>password</code></td>
+                <td><code>"admin"</code></td>
+                <td>The password used for authentication prior to publishing events via binary publisher</td>
+            </tr>
+        </tbody>
+    </table>
+    
+    The binary receiver URL/URLs need to be added as an Array using the key \[\[throttlingConfig.binary.URLGroup\]\].
```
Comment (praminda) [inline en/docs/how-tos/rate-limiting/distributed-throttling.md:129]:
```suggestion
    The binary receiver URL(s) need to be added as an Array using the key \[\[throttlingConfig.binary.URLGroup\]\]. 
```

File: en/docs/how-tos/rate-limiting/distributed-throttling.md (hunk 2 above)
Comment (praminda) [reply en/docs/how-tos/rate-limiting/distributed-throttling.md:106]:
```suggestion
    The message broker connection URL. For e.g. a [WSO2 API instance can be used as the Traffic Manager](https://apim.docs.wso2.com/en/latest/install-and-setup/setup/distributed-deployment/product-profiles/). 
```

File: en/docs/how-tos/rate-limiting/distributed-throttling.md [hunk 4]
```diff
@@ -1,14 +1,30 @@
...
+supports distributed throttling where it is able to work with a central traffic management solution. In this case, 
+multiple microgateways can connect with WSO2 API Manager 
+([WSO2 Traffic Manager](https://apim.docs.wso2.com/en/latest/install-and-setup/setup/distributed-deployment/product-profiles/)) 
+and perform rate-limiting precisely.
+
+!!! note
+    The WSO2 Traffic Manager refers to the WSO2 API Manager running in 
+    [Traffic Manager Profile](https://apim.docs.wso2.com/en/latest/install-and-setup/setup/distributed-deployment/product-profiles/). 
+    If you start the WSO2 API Manager without providing any profile, it runs as All in One Node (All the profiles 
+    are activated). For testing purposes, you can simply start the API Manager following quick start guide and test
```
Comment (AmaliMatharaarachchi) [inline en/docs/how-tos/rate-limiting/distributed-throttling.md:19]: "If you start the WSO2 API Manager without providing any profile, it runs as All in One Node (All the profiles are activated). For testing purposes, you can simply start the API Manager following quick start guide and test" I suppose we can change the sentence/ change the order to make this less confusing?

File: en/docs/how-tos/rate-limiting/distributed-throttling.md [hunk 5]
```diff
@@ -22,40 +38,157 @@ The API Microgateway upon recieving a request, checks against the local counter
            Project 'petstore' is initialized successfully.
            ```
 
-2.  Now lets add the API(open API definition) to the project. Navigate to the `           /petstore/api_definitions          ` directory. Add the API definition(s) to th `           is          ` directory. A sample open API definition can be found [here](https://github.com/wso2/product-microgateway/blob/master/samples/petstore_basic.yaml) .
+2.  Now let's add the API(open API definition) to the project. Navigate to the `/petstore/api_definitions` directory. 
+    Add the API definition(s) to this directory. A sample open API definition can be found 
+    [here](https://github.com/wso2/product-microgateway/blob/master/samples/petstore_basic.yaml). To provide an API 
+    Level throttling tier, add the following extension to the openAPI in root level.
```
Comment (AmaliMatharaarachchi) [inline en/docs/how-tos/rate-limiting/distributed-throttling.md:44]:
```suggestion
    Level throttling tier, add the following extension to the openAPI at the root level.
```

File: en/docs/how-tos/rate-limiting/distributed-throttling.md [hunk 6]
```diff
@@ -22,40 +38,157 @@ The API Microgateway upon recieving a request, checks against the local counter
...
 
     ``` java tab="Format"
     micro-gw build <project_name>
     ```
 
     ``` java tab="Example"
-    $ micro-gw build petstore
-      Build successful  for the project - petstore
+    micro-gw build petstore
+    Build successful  for the project - petstore
```
Comment (AmaliMatharaarachchi) [inline en/docs/how-tos/rate-limiting/distributed-throttling.md:62]: Do we need this line for the example? 
```suggestion
```

File: en/docs/how-tos/rate-limiting/distributed-throttling.md [hunk 7]
```diff
@@ -22,40 +38,157 @@ The API Microgateway upon recieving a request, checks against the local counter
...
 
     Once the above command is executed, An executable file ( `/petstore/target/petstore.jar` ) is created to expose the API via WSO2 API Microgateway
 
-4.  Navigate to the `<MGW-RUNTIME-HOME>/conf/` directory and open the “micro-gw.conf" file.
-5.  Enable the `enabledGlobalTMEventPublishing` property found inside the `throttlingConfig` tag. This will allow the API Microgateway to connect with the central traffic manager.
+5.  Navigate to the `<MGW-RUNTIME-HOME>/conf/` directory and open the “micro-gw.conf" file.
+
+6.  Enable the `enabledGlobalTMEventPublishing` property found inside the `throttlingConfig` tag. This will allow the API Microgateway to connect with the central traffic manager.
 
     ``` java
```
Comment (AmaliMatharaarachchi) [inline en/docs/how-tos/rate-limiting/distributed-throttling.md:71]:
```suggestion
    ``` toml
```

File: en/docs/how-tos/rate-limiting/distributed-throttling.md [hunk 8]
```diff
@@ -22,40 +38,157 @@ The API Microgateway upon recieving a request, checks against the local counter
...
+        </tbody>
+    </table>
+
+8.  If you are using API Manager 3.2.0 or later version, you need to configure the API Manager Eventhub. This step is 
+    not required if you need to do only the resource level throttling or API level throttling.
+    <!---TODO:@VirajSalaka Add the Event hub configuration guide URL--->
+    
+9.  Finally the added configurations in `microgw.conf` would look like this. 
+    
+    ``` text tab="Sample Configuration"
```
Comment (AmaliMatharaarachchi) [inline en/docs/how-tos/rate-limiting/distributed-throttling.md:169]:
```suggestion
    ``` toml tab="Sample Configuration"
```

File: en/docs/how-tos/rate-limiting/distributed-throttling.md (hunk 4 above)
Comment (VirajSalaka) [reply en/docs/how-tos/rate-limiting/distributed-throttling.md:19]: yes. I will make the change.

Comment (VirajSalaka) [reply en/docs/how-tos/rate-limiting/distributed-throttling.md]: updated via https://github.com/wso2/docs-mg/pull/41/commits/dae05bde11563c05e47375d5d63004a64b7ea4e4

File: en/docs/how-tos/rate-limiting/distributed-throttling.md [hunk 9]
```diff
@@ -22,40 +36,156 @@ The API Microgateway upon recieving a request, checks against the local counter
            Project 'petstore' is initialized successfully.
```
Comment (AmaliMatharaarachchi) [inline en/docs/how-tos/rate-limiting/distributed-throttling.md:36]:
```suggestion
```

**Review by AmaliMatharaarachchi — APPROVED** [review]

**Pull request merged**